src/python/pants/help/help_printer.py

```diff
@@ -47,6 +51,23 @@ def print_hint():
       return 1
     return 0
 
+  def _print_goals_help(self):
+    goal_descriptions = {}
+    for scope_info in self._options.known_scope_to_info.values():
+      if scope_info.category not in (ScopeInfo.GOAL, ScopeInfo.TASK):
```

Comment:
Why is `ScopeInfo.TASK` included here? FWICT only v1 `TaskBase` tasks will have this scope and those tasks will be registered in v1 Goals discovered in `Goal.all()` below.

Reply:
Yea, true. `Goal` goes hunting for Task docstrings where need be. before/after diff looks the same.